tuner/examples/test/tuner_test.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2024 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""
+Sample Usage:
+
+python -m examples.test double_mmt.mlir mmt_benchmark.mlir --devices=hip://0,hip://1 --num-candidates=64
+
+
+Recommended Trial Run:
+
+python -m examples.test double_mmt.mlir mmt_benchmark.mlir --num-candidates=10
+
+
+Dry Run Test (no gpu required, not currently working):
+
+python -m examples.test double_mmt.mlir mmt_benchmark.mlir --num-candidates=64 --dry-run
+
+"""
+
+from tuner import libtuner
+from pathlib import Path, PurePath
+import argparse
+import os
+
+
+class TestTuner(libtuner.TuningClient):
+    def __init__(self):
+        self.compile_flags = [
+            "--iree-hip-target=gfx942",
+            "--compile-from=executable-sources",
+        ]
+        self.benchmark_flags = [
+            "--benchmark_repetitions=3",
+            "--benchmark_format=json",
```

Comment:
Eventually we should use the bindings for benchmarking too. Just FYI, this is definitely outside of the scope of this PR.

Reply:
Yeah, that's actually what I did at first, but the compiler bindings did not seem to recognize many of the compiler codegen flags. I didn't dig too deep into it, but I think we may need to expose more flags to the compiler api in IREE before we can do this for compilation. And then since I was already using subprocess for compilation, I just did the same for benchmarking for consistency.